pandas/tests/test_nanops.py

```diff
@@ -327,6 +327,12 @@ def test_nansum(self):
         self.check_funs(nanops.nansum, np.sum, allow_str=False,
                         allow_date=False, allow_tdelta=True, check_dtype=False)
 
+    def test_nansum_overflow(self):
+        s = Series([2**31])
```

Comment:
needs testing both with and w/o bottleneck (to yield the same results), see other tests for how to do this.